src/Faker/Container/Container.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Faker\Container;
+
+use Psr\Container\ContainerInterface;
+
+/**
+ * A simple implementation of a container.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ * @experimental This class is experimental and does not fall under our BC promise
+ */
+final class Container implements ContainerInterface
+{
+    /**
+     * @var array<string, callable|string|object>
+     */
+    private $definitions = [];
+    private $services = [];
+
+    public function __construct(array $container)
+    {
+        $this->definitions = $container;
+    }
+
+    /**
+     * @throws NotInContainerException
+     * @throws ContainerException
+     */
+     public function get($id)
+    {
+        if (!is_string($id)) {
+            throw new \LogicException(sprintf('First argument of %s::get() must be string', __CLASS__));
+        }
+
+        if (isset($this->services[$id])) {
+            return $this->services[$id];
+        }
+
+        if (!$this->has($id)) {
+            throw new NotInContainerException(sprintf('There is not service with id "%s" in the container.', $id));
+        }
+
+        $definition = $this->definitions[$id];
+        if (is_string($definition)) {
+            if (!class_exists($definition)) {
+                throw new ContainerException(sprintf('Could not instantiate class "%s". Class was not found.', $id));
+            }
+
+            try {
+                return $this->services[$id] = new $definition();
```

Comment:
Can we reduce the scope of the `try` block to things that can go wrong?

Things that can't go wrong:

- assigning a value
- returning a value

Yes, it's more code, but

- we clarify intent
- we avoid confusing statements

Reply:
I thought this method was long enough already, That is why I wrote it in a compact way.